Source/WebCore/Modules/gamepad/NavigatorGamepad.cpp

```diff
@@ -89,8 +92,46 @@ ExceptionOr<const Vector<RefPtr<Gamepad>>&> NavigatorGamepad::getGamepads(Naviga
     return NavigatorGamepad::from(navigator)->gamepads();
 }
 
+// The UIProcess tracks when a WebPage has recently used gamepads to configure certain behaviors on the page.
+//
+// Once a WebPage notifies the UIProcess that gamepads have been accessed, the UIProcess starts a timer with
+// a short delay, after which it assumes that WebPage is no longer actively using gamepads.
```

Comment:
I didn't even recall `HysteresisActivity` - Neat!

Its behavior is a perfict fit for the UIProcess-side of this, and I've made that change.

Start/Stop from the web process doesn't quite work.
The Web Process side still needs to periodically send the "I'm still using them" heartbeat to the UI process to drive the `impulse()` of its hysteresis.

Reply:
Brady explained offline that the main reason we prefer heartbeat mechanism instead of having web process sending"start-stop" messages is that UI process needs to make adjustment when web process hangs or is blocked (e.g. on sync message) and cannot send stop message.